### ISSUE TITLE
Catplot: Allow custom values for the confidence interval

### ISSUE THREAD

**Issue by kaisengit** [mwaskom/seaborn#1667]
Hello, I am plotting a catplot like this:

`
sns.catplot(data=df, x=exp, y=sum_total, col=level, kind='point', col_wrap=5, ci='sd', sharey=False, hue=domain, sharex=False)
`

As you can see I use the standard deviation for the CI. Therefore I pass several observations (replicates) of a single event to seaborn and it computes and plots the CI via the SD. Now for non-log data this is fine.
However, I work in intensity ranges where only a log scale makes sense. I know it is possible to set a log scale on the axes but this is not what I want. I want to plot the log2 of my data since it is much easier to see when intensities double. 

The following is of course possible:
`
df[sum_total] = np.log2(df[sum_total])
`
I can pass this to the catplot. However, it will then compute the mean and the standard deviation using the log2 values which is not the same as computing the mean and the standard deviation beforehand and the computing the log2 values.

To sum it up: I'd like to have way to visualize the log2 values of my data but still have the mean and SD computed correctly. I do not want a logarithmic scale my axis. So either I'd need to calculate the SD and mean on my own, take the log2 and pass that to seaborn (which is afaik not possible) OR I'd need a way to convert all my y-values to their log2s after seaborn has plotted (and re-scale the axes). I am not really sure whether this a bug report or a feature request but I'd be grateful for any input!

**Comment by mwaskom** [comment]
> I know it is possible to set a log scale on the axes but this is not what I want. I want to plot the log2 of my data since it is much easier to see when intensities double.

I don't follow this. You can control the base of the log scale:

```
x = np.arange(10) + 1
y = 2 ** x
f, ax = plt.subplots()
ax.plot(x, y)
ax.set_yscale("log", basey=2)
```
![image](https://user-images.githubusercontent.com/315810/52140942-31566580-2622-11e9-8984-96c57083e7cf.png)

Specifying values for the confidence intervals isn't an option and will not be added.

**Comment by kaisengit** [comment]
In a convoluted way this was exactly what I've been looking for. Thank you!
